webcrypto/subtle_crypto.go

```diff
@@ -383,9 +437,56 @@ func (sc *SubtleCrypto) ImportKey(
 //
 // The `format` parameter identifies the format of the key data.
 // The `key` parameter is the key to export, as a CryptoKey object.
+//
+// TODO @oleiade: implement support for JWK format
 func (sc *SubtleCrypto) ExportKey(format KeyFormat, key goja.Value) *goja.Promise {
-	// TODO
-	return nil
+	promise, resolve, reject := sc.makeHandledPromise()
+
+	var algorithm Algorithm
+	algValue := key.ToObject(sc.vu.Runtime()).Get("algorithm")
+	if err := sc.vu.Runtime().ExportTo(algValue, &algorithm); err != nil {
+		reject(NewError(0, SyntaxError, "key is not a valid CryptoKey"))
+		return promise
+	}
+
+	// 5.
+	if !isRegisteredAlgorithm(algorithm.Name) {
+		reject(NewError(0, NotSupportedError, "unsupported algorithm "+algorithm.Name))
+		return promise
```

Comment:
If we're being strict about following [the spec](https://www.w3.org/TR/WebCryptoAPI/#dfn-SubtleCrypto-method-exportKey), everything after step 3 should be done asynchronously.

Reply:
Good catch 🙇🏻